### PR TITLE
nixos/zram: use zram-generator

### LINKED ISSUE #210161: packaging request: zram-generator
**Project description**

In https://github.com/NixOS/nixpkgs/pull/102855 the author says,

> In future it might be good to port this to the systemd zram-generator,
> instead of using this separate infrastructure.

In 2020 the zram-generator doesn't seem to have had any releases.  In the meantime, it's gone from initial release version 0.2.0 up to the somewhat-mature version 1.1.2. 

It's probably a good time to port this to zram-generator.

**Metadata**

* homepage URL: https://github.com/systemd/zram-generator
* source URL: https://github.com/systemd/zram-generator
* license: mit
* platforms: linux

### PR DESCRIPTION
**Pull request by NickCao**
###### Description of changes
Followup of https://github.com/NixOS/nixpkgs/pull/210190
Closes https://github.com/NixOS/nixpkgs/issues/210161

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by NickCao** [comment]
The current approach preserved the original module interface as much as possible. While another approach would be revamping the module to bring it closer to what zram-generator expects.

**Comment by NickCao** [comment]
@ofborg test zram-generator

File: nixos/tests/zram-generator.nix [hunk 1]
```diff
@@ -2,12 +2,14 @@ import ./make-test-python.nix {
   name = "zram-generator";
 
   nodes.machine = { pkgs, ... }: {
```
Comment (flokli) [inline nixos/tests/zram-generator.nix:4]: You don't need `pkgs` anymore.

**Review by flokli — APPROVED** [review]
small nit, LGTM otherwise.

**Comment by NickCao** [comment]
@ofborg test zram-generator

**Review by flokli — APPROVED** [review]

**Pull request merged**